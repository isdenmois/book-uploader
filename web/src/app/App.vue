<script setup lang="ts">
import SearchPage from 'pages/search-page.vue'
import LoginPage from 'pages/login-page.vue'
import { useAuth } from 'features/auth'
import { useFilters } from 'features/filters'

const auth = useAuth()
const filters = useFilters()
let showLogin = $computed(() => filters.source === 'ZLIB' && !auth.cookie)
</script>

<template>
  <component :is="showLogin ? LoginPage : SearchPage" />
</template>

<style>
#app {
  height: 100vh;
  overflow: hidden;
  display: flex;
  flex-direction: column;
}
</style>
