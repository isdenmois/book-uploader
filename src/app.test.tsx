import { render } from '@testing-library/react-native';
import React from 'react';

import 'react-native-gesture-handler/jestSetup';
jest.mock('@react-native-community/async-storage', () => ({ getItem: () => null }));
jest.mock('@react-navigation/stack', () => ({
  createStackNavigator: () => ({
    Navigator({ children, ...props }) {
      const Element: any = 'Navigator';

      return <Element {...props}>{children}</Element>;
    },
    Screen(props) {
      const Element: any = 'Screen';

      return <Element {...props} />;
    },
  }),
}));
jest.mock('@react-navigation/native', () => ({
  NavigationContainer({ children, ...props }) {
    const Element: any = 'NavigationContainer';

    return <Element {...props}>{children}</Element>;
  },
}));
jest.mock('@react-native-community/async-storage', () => ({}));
jest.mock('react-native-fs', () => ({}));

import App from './app';

test('App', () => {
  const { toJSON } = render(<App />);

  expect(toJSON()).toMatchSnapshot();
});
