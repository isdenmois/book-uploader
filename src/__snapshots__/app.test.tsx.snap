// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<NavigationContainer>
  <Navigator
    screenOptions={
      Object {
        "cardStyle": Object {
          "backgroundColor": "#F9FAFE",
        },
      }
    }
  >
    <Screen
      component={[Function]}
      name="main"
      options={
        Object {
          "header": [Function],
        }
      }
    />
    <Screen
      component={[Function]}
      name="scan"
      options={
        Object {
          "title": "",
        }
      }
    />
  </Navigator>
</NavigationContainer>
`;
